src/shared_utilities/models.jl

```diff
@@ -447,6 +447,26 @@ function initialize(model::AbstractModel{FT}) where {FT}
     return Y, p, coords
 end
 
+
+"""
+    initialize_from_checkpoint(restart_file; model::AbstractModel)
+
+Creates the prognostic and auxiliary states structures, but with unset
+values; constructs and returns the coordinates for the `model` domain.
+We may need to consider this default more as we add diverse components and
+`Simulations`.
+
```

Comment:
Maybe we could add to the docstring why model is optional. 

and, what happens if the user saves the checkpoint with the model hash, but does not pass the mode in here? (maybe youve handled this already, I havent gotten there yet! if so, ignore)

Reply:
ok - nothing is done in that case. that's fine!